### PR TITLE
utilccl: don't require the org name in checkLicense

### PR DESCRIPTION
**Pull request by andreimatei**
The license checking functions required the org name to be passed in, even though it was also taking the settings and the org name comes from a setting. This patch removes the need to pass in that arg. This aims to make it a bit easier for random call sites to check for a license, which currently is a bit tedious.

The change is in license_check.go. Everything else is removing the args from cascading callers.

Release note: None
Epic: None

### CODE REVIEW

File: pkg/ccl/utilccl/BUILD.bazel [hunk 1]
```diff
@@ -15,6 +15,7 @@ go_library(
         "//pkg/server",
         "//pkg/settings",
         "//pkg/settings/cluster",
+        "//pkg/sql",
```
Comment (dt) [inline pkg/ccl/utilccl/BUILD.bazel:18]: This a hefty dependency for a util pkg; I wonder if we could move that setting registration somewhere less expensive to depend on

**Review by dt — APPROVED** [review]

File: pkg/ccl/utilccl/BUILD.bazel (hunk 1 above)
Comment (andreimatei) [reply pkg/ccl/utilccl/BUILD.bazel:18]: do you feel this way even though the pig already depends on `server` ?

Comment (srosenberg) [reply pkg/ccl/utilccl/BUILD.bazel:18]: "The pig" intrigues me :) Is that a typo or does it stand for something other than "package in question"?

Comment (dt) [reply pkg/ccl/utilccl/BUILD.bazel:18]: I assume autocorrect of pkg

**Review by andreimatei — COMMENTED** [review]
TFTRs!

bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/91768#-:-NGrRi0G4tMYl69rWkBf:b-tlx7ko)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner and @srosenberg)

**Pull request merged**